### DISCUSSION TITLE
request.client broken in version 0.35.0?

### DISCUSSION THREAD

**Discussion by jsoques** [encode/starlette#2408] [Potential Issue] [answered]
I have been using request.client.host previously and when upgraded to 0.35.0 client is always 'None'. I have been using it to obtain the clients address as indicated here: https://www.starlette.io/requests/. I have reverted back to version 0.32.0.post1. This occurs when using TestClient only apparently so my tests are broken (testing FastAPI project).

**Comment by Kludex** [comment, marked as answer]
That is the intended behavior.

**Reply by VenturaParade** [reply]
If anyone got here wondering why or how to fix, I _think_ [this](https://github.com/encode/starlette/issues/2295) is the cause.
An easy fix is just to do `request.client.host if request.client else None`, since this behavior is exclusive to the test client

**Reply by pythonweb2** [reply]
@Kludex can you make it optional to set the test client to None? We have a lot of tests that relied on it being populated for logging. ☹️

**Reply by blaise-io** [reply]
It's reverted (for now), see https://github.com/encode/starlette/pull/2525

**Reply by pythonweb2** [reply]
This change was in `0.37.2`
